Display a custom page for Dev Tools Discovery

WizardCM opened the pull request:
### Description

This enables a custom resource handler for CEF to allow overriding of individual internal files that CEF attempts to load.

The Dev Tools Discovery page is exposed via Remote Debugging and provides a list of all loaded browser pages to easily open Dev Tools from a single location.

This updated version introduces a number of enhancements:
1. Items are now ordered by category
2. Sub-pages are listed under parent pages
3. Built-in refresh button
4. OBS version number in page body based on User Agent
5. Clicking items opens them in a new tab by default
6. Custom styling

**Before:**

![image](https://user-images.githubusercontent.com/941350/227149773-a530af97-6bb0-4b12-b152-7ac528c49640.png)

**After:**

![image](https://user-images.githubusercontent.com/941350/227149688-03811902-071b-49fc-986b-5c32853a3317.png)

### Motivation and Context

This file is normally packaged within CEF builds themselves. It's unnecessary to rebuild CEF just to change one HTML file.

Remote debugging is used for development and troubleshooting of browser sources.

### How Has This Been Tested?

Launch OBS with `--remote-debugging-port=1234` and visit the page.

### Types of changes
 - New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

gxalpha commented on obs-browser-plugin.cpp:
```diff
@@ -388,6 +388,12 @@ static void BrowserInit(void)
 	app = new BrowserApp(tex_sharing_avail);
 
+	char *pagesPath = obs_module_file("dev_tools_discovery.html");
+	char *pagesFile = os_quick_read_utf8_file(pagesPath);
```
Could these be BPtr's instead (making the explicit bfree calls unneeded)?

WizardCM replied:
Ah, I always forget about BPtr. Yes, that works here.

norihiro commented:
The two lines of the commit description have 73 characters.
```
The Dev Tools Discovery page is exposed via Remote Debugging and provides
a list of all loaded browser pages to easily open Dev Tools from a single
```

WizardCM commented:
Fixed commit description, and based on Warchamp's feedback off-thread I've removed the collapse capability as it's unnecessary.

Should be good to merge.

Lain-B commented:
Going to revert this for now based upon more feedback from Warchamp. You might have to remake this Matt. Sorry for the trouble. Also I should have spotted that you were using a raw pointer -- definitely want to use RAII and not manually free memory in the future. An std::string would have been sufficient for the `devToolsFile` variable.
